### ISSUE TITLE
Sample Weighted BatchNorm1d

### ISSUE THREAD

**Issue by jonathanBalasingham** [pytorch/pytorch#89724]
### 🚀 The feature, motivation and pitch

I recently worked on a project involving graph neural networks where batch normalization with weighted samples (nodes) helped improve prediction performance. This was especially useful in my use case where graphs could vary quite significantly in size and weights reduce influence of larger graphs (those having more nodes) within the batch. Weights can also be used to compress graphs, improving performance. I don't believe there is a way currently to pass sample weights to the batch norm layer, so I wrote my own. This version of course, doesn't have the performance of the original BatchNorm1d layer. I was wondering whether it would be worthwhile for me to implement this directly. 

### Alternatives

The only feasible alternative I could come up with was repeating samples such that the behavior was replicated through the current BatchNorm1d i.e. if a sample contained [0, 1] with respective weights [1/3, 2/3] the sample could be expanded to [0, 1, 1]. Though, this is obviously inefficient.

### Additional context

There is a weighted form of Welford's algorithm: [https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Weighted_incremental_algorithm](url), so performance can likely be retained (though this will of course be tested).
